src/instructlab/eval/mt_bench_common.py

```diff
@@ -273,13 +294,20 @@ def chat_completion_openai(
             output = response.choices[0].message.content
             break
         except openai.OpenAIError as e:
+            if _is_fatal_openai_error(e):
```

Comment:
I might not be following this correctly yet.  The logic I was expecting was:

400 - Include in error rate after 1 occurrence
422-500 - Include in error rate after retries
401-404 + APIConnectionError - fail after retries

If we are only trying to handle APIConnectionError, I am not quite following why APIConnectionErrors are being allowed to continue but timeout errors and all other errors seem to be marked as non fatal but will exit below when their output is none.

Reply:
> But I think **422** is indicative of the input not being accepted. MDN [suggests](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422) that repeating it won't produce a different result. Should we handle it like 400 then? (meaning: Include in error rate.)

+1, good correction

> 
> > 401-404 - fail after retries.
> 
> What's the point of retrying `AuthenticationError` or `PermissionDeniedError` or `NotFoundError` (especially the last two). Don't we expect that if access is denied, then it will be denied if we retry again? (Same for 404.)

I am iffy on these between fail fast and fail after retries.  My fear was there could be flakiness in these responses.  In which case the only harm that's done with retries is it would wait for 20ish seconds before it declared it a failure and gave up.

Reply:
OK I will include these in retry list, with a note that we may explore later if they justify retries.